### PR TITLE
test: delete board service test

### PR DESCRIPTION
**Pull request by GoncaloCanteiro**
Relates to:
- #1228 

## Proposed Changes

  - Add delete.board.service.spec.ts
  - Add unit tests for the delete method
  - Add unit tests for the deleteBoardsByTeamId method

This pull request closes:
- #1228

### CODE REVIEW

**Review by nunocaseiro — CHANGES_REQUESTED** [review]
GJ here, you have just missed one test for the 40th line

image

File: backend/src/modules/boards/services/delete.board.service.spec.ts [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+			boardRepositoryMock.deleteBoard.mockResolvedValue(board);
+
+			deleteSchedulesServiceMock.deleteScheduleByBoardId.mockResolvedValue(null);
+			await boardRepositoryMock.deleteManySubBoards.mockResolvedValue(2);
+			await deleteBoardUserServiceMock.deleteDividedBoardUsers.mockResolvedValue(2);
+			await deleteBoardUserServiceMock.deleteSimpleBoardUsers.mockResolvedValue(2);
+
+			//Slack Enabled
+			await boardRepositoryMock.getBoardPopulated.mockResolvedValue(board);
+			await achiveChannelServiceMock.execute.mockResolvedValue(null);
```
Comment (mourabraz) [inline backend/src/modules/boards/services/delete.board.service.spec.ts:79]: I believe you should not use "await" after "mockResolvedValue". Does not make sense to return a Promise when you build a mock, this is only an arrangement for when the code being tested runs.

File: backend/src/modules/boards/services/delete.board.service.spec.ts [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+		it('should throw error when deleteBoardBoardUsersAndSchedules fails', async () => {
+			const board = BoardFactory.create({
+				dividedBoards: BoardFactory.createMany(2),
+				slackEnable: true
+			});
+			boardRepositoryMock.getBoard.mockResolvedValue(board);
+			boardRepositoryMock.deleteBoard.mockResolvedValue(board);
+			deleteSchedulesServiceMock.deleteScheduleByBoardId.mockResolvedValue(null);
+			await boardRepositoryMock.deleteManySubBoards.mockResolvedValue(1);
+			await deleteBoardUserServiceMock.deleteDividedBoardUsers.mockResolvedValue(2);
```
Comment (mourabraz) [inline backend/src/modules/boards/services/delete.board.service.spec.ts:99]: The same here. You are building a mock not executing the conde...

File: backend/src/modules/boards/services/delete.board.service.spec.ts [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+
+			await boardRepositoryMock.deleteManySubBoards.mockResolvedValue(2);
+			await deleteBoardUserServiceMock.deleteDividedBoardUsers.mockResolvedValue(2);
+			await deleteBoardUserServiceMock.deleteSimpleBoardUsers.mockResolvedValue(2);
+
+			//Slack Enabled
+			await boardRepositoryMock.getBoardPopulated.mockResolvedValue(board);
+			await achiveChannelServiceMock.execute.mockResolvedValue(null);
+
+			await expect(service.deleteBoardsByTeamId('teamId')).resolves.toBe(true);
```
Comment (mourabraz) [inline backend/src/modules/boards/services/delete.board.service.spec.ts:124]: and here

File: backend/src/modules/boards/services/delete.board.service.spec.ts [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+			boardRepositoryMock.deleteBoard.mockResolvedValue(board);
+			deleteSchedulesServiceMock.findAndDeleteScheduleByBoardId.mockResolvedValue(null);
+
+			await boardRepositoryMock.deleteManySubBoards.mockResolvedValue(2);
+			await deleteBoardUserServiceMock.deleteDividedBoardUsers.mockResolvedValue(2);
+			await deleteBoardUserServiceMock.deleteSimpleBoardUsers.mockResolvedValue(2);
+
+			//Slack Enabled
+			await boardRepositoryMock.getBoardPopulated.mockResolvedValue(board);
+			await achiveChannelServiceMock.execute.mockResolvedValue(null);
```
Comment (mourabraz) [inline backend/src/modules/boards/services/delete.board.service.spec.ts:122]: the same as before

File: backend/src/modules/boards/services/delete.board.service.spec.ts [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+			boardRepositoryMock.deleteBoard.mockResolvedValue(board);
+			deleteSchedulesServiceMock.findAndDeleteScheduleByBoardId.mockResolvedValue(null);
+
+			await boardRepositoryMock.deleteManySubBoards.mockResolvedValue(0);
+			await deleteBoardUserServiceMock.deleteDividedBoardUsers.mockResolvedValue(0);
+			await deleteBoardUserServiceMock.deleteSimpleBoardUsers.mockResolvedValue(0);
+
+			//Slack Enabled
+			await boardRepositoryMock.getBoardPopulated.mockResolvedValue(board);
+			await achiveChannelServiceMock.execute.mockResolvedValue(null);
```
Comment (mourabraz) [inline backend/src/modules/boards/services/delete.board.service.spec.ts:143]: same here

File: backend/src/modules/boards/services/delete.board.service.spec.ts (hunk 1 above)
Comment (GoncaloCanteiro) [reply backend/src/modules/boards/services/delete.board.service.spec.ts:79]: That's a good point, thank you for the tips! 👍

File: backend/src/modules/boards/services/delete.board.service.ts [hunk 6]
```diff
@@ -35,7 +35,7 @@ export default class DeleteBoardService implements DeleteBoardServiceInterface {
 		}
 
 		try {
-			return this.deleteBoardBoardUsersAndSchedules(boardId, true);
+			return await this.deleteBoardBoardUsersAndSchedules(boardId, true);
```
Comment (nunocaseiro) [inline backend/src/modules/boards/services/delete.board.service.ts:38]:
```suggestion
			return this.deleteBoardBoardUsersAndSchedules(boardId, true);
```

**Review by nunocaseiro — APPROVED** [review]
LGTM 🚀 you can wait for Carlo's review

**Pull request merged**